src/test/kotlin/org/rust/lang/core/resolve/RsPathCachingRootTest.kt

```diff
@@ -88,32 +89,40 @@ class RsPathCachingRootTest : RsTestBase() {
                //X  //^  //^ //^ //^
     """)
 
-    // TODO move the caching root upper
     fun `test path arguments with fn pointer`() = doTest("""
         struct Foo<T>(T);
         struct Bar;
         type T = Foo<fn(Foo<Foo<Foo<Bar>>>) -> Foo<Foo<Foo<Bar>>>>;
```

Comment:
Lets also add test with multiple `fn` arguments:
```rust
        struct Foo {}
        type F = fn(  Foo, Foo);
               //X  //^  //^
```
And probably it is worth to support usual functions arguments in future
```rust
        struct Foo {}
        fn func(x: Foo, y: Foo) {}
             //X //^     //^
```

Reply:
>Lets also add test with multiple fn arguments

✅

> And probably it is worth to support usual functions arguments in future

I don't want to do it in this PR, but I specially added tests for this case for future improvements